wave_kinematics/include/wave/kinematics/rotation.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef __WAVE_KINEMATICS_ROTATION_HPP__
+#define __WAVE_KINEMATICS_ROTATION_HPP__
+
+#include <iostream>
+#include <functional>
+
+#include "wave/utils/utils.hpp"
+#include <kindr/Core>
+
+namespace wave {
+
+class Rotation {
+public:
+  // Internal storage as a kindr Rotation object.
+  kindr::RotationMatrixD rotation_object;
+  // The comparision threshold used to test if rotation objects are
+  // "close enough" to each other.  Used in unit testing and the "isNear"
+  // function.
+  static double comparision_threshold;
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  // Rotation Object Initializers.
+  Rotation();
+  static bool fromEulerXYZ(const double rotation_x,
```

Comment:
So with these prototypes, does it change the object in-place, and also return a ref to that object? or a ref to a new object, but doesn't touch "this" object?